doc/user/content/sql/create-source/postgres.md

```diff
@@ -152,7 +152,46 @@ ingestion progress and debugging related issues, see [Troubleshooting](/ops/trou
 
 ##### Supported types
 
-Replicating tables that contain [data types](/sql/types/) unsupported in Materialize is possible via the `TEXT COLUMNS` option. The specified columns will be treated as `text`, and will thus not offer the expected PostgreSQL type features. For example:
+Materialize natively supports the following PostgreSQL types:
+
+<ul style="column-count: 3">
+<li><code>bool</code></li>
+<li><code>bpchar</code></li>
+<li><code>bytea</code></li>
+<li><code>char</code></li>
+<li><code>date</code></li>
+<li><code>daterange</code></li>
+<li><code>float4</code></li>
+<li><code>float8</code></li>
+<li><code>int2</code></li>
+<li><code>int2vector</code></li>
+<li><code>int4</code></li>
+<li><code>int4range</code></li>
+<li><code>int8</code></li>
+<li><code>int8range</code></li>
+<li><code>interval</code></li>
+<li><code>json</code></li>
```

Comment:
Do we support `json` natively? I thought we were just falling back to `jsonb` under the hood. Just trying to grok how you came to a list including `json` and `timez`, so I know where and how to look in the future. 🙇‍♀️

Reply:
I looked here: https://github.com/MaterializeInc/materialize/blob/c8a8ee7c0542d9f40e5afc34ee2606db2d790da9/src/pgrepr/src/types.rs#L546-L654

But I forgot to account for the fact that `timetz` is later converted to an error here: https://github.com/MaterializeInc/materialize/blob/c8a8ee7c0542d9f40e5afc34ee2606db2d790da9/src/pgrepr/src/types.rs#L1004-L1109

We do support `json` natively, but only because of this giant special case: https://github.com/MaterializeInc/materialize/blob/c8a8ee7c0542d9f40e5afc34ee2606db2d790da9/src/sql/src/plan/statement.rs#L781-L786

tl;dr I think the best thing to do is to maintain this list by hand; there's no one place in the codebase that's authoritative, and some of the support/nonsupport is quite subtle. https://github.com/MaterializeInc/materialize/pull/26107 just removed support for the reg* types, for example. PR incoming for that shortly.